src/ui/public/es.js

```diff
@@ -9,38 +9,39 @@ import 'elasticsearch-browser';
 import _ from 'lodash';
 import uiModules from 'ui/modules';
 
-let es; // share the client amongst all apps
+const plugins = [function (Client, config) {
+  // esFactory automatically injects the AngularConnector to the config
+  // https://github.com/elastic/elasticsearch-js/blob/master/src/lib/connectors/angular.js
+  class CustomAngularConnector extends config.connectionClass {
+    request = _.wrap(this.request, function (request, params, cb) {
+      if (String(params.method).toUpperCase() === 'GET') {
+        params.query = _.defaults({ _: Date.now() }, params.query);
+      }
+
+      return request.call(this, params, cb);
+    });
+  }
+
+  config.connectionClass = CustomAngularConnector;
+}];
+
 uiModules
   .get('kibana', ['elasticsearch', 'kibana/config'])
-  .service('es', function (esFactory, esUrl, $q, esApiVersion, esRequestTimeout) {
-    if (es) return es;
-
-    es = esFactory({
+  .service('es', function (esFactory, esUrl, esApiVersion, esRequestTimeout) {
```

Comment:
The problem with making it smarter is plugins - with core kibana we know it's only the .kibana index, but with x-pack we have .monitoring-*, and more.  Maybe plugins could register administration indices.

 👍 comment being added, and I'll make sure it's in the elastic.co plugin docs too